table/table_test.go

```diff
@@ -367,6 +409,326 @@ func (t *TableWritingTestSuite) TestAddFilesFailsSchemaMismatch() {
 `)
 }
 
+func (t *TableWritingTestSuite) TestAddFilesPartitionedTable() {
+	ident := table.Identifier{"default", "partitioned_table_v" + strconv.Itoa(t.formatVersion)}
+	spec := iceberg.NewPartitionSpec(
+		iceberg.PartitionField{SourceID: 4, FieldID: 1000, Transform: iceberg.IdentityTransform{}, Name: "baz"},
+		iceberg.PartitionField{SourceID: 10, FieldID: 1001, Transform: iceberg.MonthTransform{}, Name: "qux_month"})
+
+	tbl := t.createTable(ident, t.formatVersion,
+		spec, t.tableSchema)
+
+	t.NotNil(tbl)
+
+	dates := []string{
+		"2024-03-07", "2024-03-08", "2024-03-16", "2024-03-18", "2024-03-19",
+	}
+
+	files := make([]string, 0)
+	for i := range 5 {
+		filePath := fmt.Sprintf("%s/partitioned_table/test-%d.parquet", t.location, i)
+		table, err := array.TableFromJSON(memory.DefaultAllocator, t.arrSchema, []string{
+			`[{"foo": true, "bar": "bar_string", "baz": 123, "qux": "` + dates[i] + `"}]`,
+		})
+		t.Require().NoError(err)
+		defer table.Release()
+
+		t.writeParquet(tbl.FS().(iceio.WriteFileIO), filePath, table)
+		files = append(files, filePath)
+	}
+
+	tx := tbl.NewTransaction()
+	t.Require().NoError(tx.AddFiles(files, nil, false))
+
+	stagedTbl, err := tx.StagedTable()
+	t.Require().NoError(err)
+	t.NotNil(stagedTbl.NameMapping())
+
+	t.Equal(stagedTbl.CurrentSnapshot().Summary,
+		&table.Summary{
+			Operation: table.OpAppend,
+			Properties: iceberg.Properties{
+				"added-data-files":        "5",
+				"added-files-size":        "3660",
+				"added-records":           "5",
+				"changed-partition-count": "1",
```

Comment:
ah okay, this is `1` instead of `5` because all the dates are in March and we have a MonthTransform